### ISSUE TITLE
Question: could we switch default projections for geographic plots ?

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#4182]
### Rationale

Currently, the `GeographicCoordinateHandler` class adds some complexity to the code base that collides with my work in #4179 

Namely, here's the branch I'm talking about:
https://github.com/yt-project/yt/blob/695646b29cf9864b286aca246724e503ac581475/yt/visualization/plot_window.py#L225-L239
This is basically the only place where the `data_projection` property of `CoordinateHandler` classes is used.
For reference, it is defined in the base class as well as in the one child class that this mechanism is relevant to, see
https://github.com/yt-project/yt/blob/695646b29cf9864b286aca246724e503ac581475/yt/geometry/coordinates/coordinate_handler.py#L185
and
https://github.com/yt-project/yt/blob/main/yt/geometry/coordinates/geographic_coordinates.py#L368

This mechanism, using ds.coordinates.data_transform and data_projection, is only used if
- coordinates are geographic
- normal axis is the "radial" axis (altitude of depth)
and it adds a hard requirement on Cartopy for this single use case.

All this is done so we can use [Mollweide projection](https://en.wikipedia.org/wiki/Mollweide_projection) by default. We *could* drop this complexity in favour of the very similar looking [Aitoff projection](https://en.wikipedia.org/wiki/Aitoff_projection)
that is already implemented natively in yt, and used for spherical geometries.

We could (and should) still support integration for cartopy projections, what I'm proposing is that we don't make it a hard requirement for default plots, and make the `GeographicCoordinateHandler` more like its sibling classes in the process.

Doing this would make allow me to implement #4179 uniformly across all geometries, the alternative being that even more complexity would need to be added to support this one special case.

**Comment by matthewturk** [comment]
@chrishavlin I'd like for you to chime in with thoughts on this

**Comment by chrishavlin** [comment]
> All this is done so we can use [Mollweide projection](https://en.wikipedia.org/wiki/Mollweide_projection) by default

Not really: the particular type of default projection does not matter matter at all. What matters is that it is a `cartopy.crs` projection and that the default plot is a cartopy `GeoAxes`. The whole point of having a GeographicCoordinateHandler relying on cartopy is so that we can have access to all the convenient methods that cartopy has implemented (adding lat/lon grids, adding coastline annotations, adding NaturalEarth features, over-plotting data in different transforms, using a different reference ellipse, etc. etc. etc.) and if you change the default to not be something that does not result in a `GeoAxes` then that would mean that yt's default geographic plot is essentially nonfunctional for geographic data. It would be a step backward in interoperability. 

That said, I'm not opposed to a refactor as long as the default behavior of plotting a geographic dataset is to return a `GeoAxes`.

As an additional note, I'm worried you are conflating transform and projection here. The GeopgraphicCoordinateHandler uses `PlateCarree` as the default `data_transform` and in #4179 you are changing this to aitoff. This is not correct. In geographic systems, the transform is the coordinate reference system on which the data is defined (https://scitools.org.uk/cartopy/docs/latest/tutorials/understanding_transform.html). This is almost always going to be `PlateCaree`: people have data in lat/lon. This should not change.

**Comment by chrishavlin** [comment]
Sorry for the delay in responding!

So I understand the impulse to have something happen for default vanilla yt. But I'd argue that using geographic coordinates in yt is **not** vanilla yt. Futhermore, installing cartopy is very standard for geoscientists, I really don't see it as a barrier to using yt.  

If you **really** want something to show up without cartopy then I suggest that rather than implementing a projection to use, it would be better to plot in what you're calling native coordinates. i.e.., when plotting at fixed radius/depth/altitude without cartopy I'd prefer to have a plot of lat/lon on cartesian axes. I'm not sure how that fits in with what you're designing, though... 

And just a final note on the "transform" vs "projection" terminology -- I've thought before that it would be clearer to rename the `data_transform` attribute to be `crs` (coordinate reference system) in the `GeopgraphicCoordinateHandler`. This is what I do in other work, it's also what geopandas does and I find it more explicit than the "transform"  terminology that cartopy uses. We'd still want to use that `crs` when as the data "transform" when initializing any cartopy GeoAxes, but given yt's use of the word transform in many other contexts, it might help with clarifying the different meanings.

**Comment by neutrinoceros** [comment]
> So I understand the impulse to have something happen for default vanilla yt. But I'd argue that using geographic coordinates in yt is not vanilla yt.

So the way I see it, if it's not vanilla yt then it should be -treatable as- an external extension. I would have no issue with cartopy being a hard requirement for such an extension.

> Futhermore, installing cartopy is very standard for geoscientists, I really don't see it as a barrier to using yt.

That's good to know, but I am coming at this from the maintainer angle. To me the problem is that some of the code supporting geographic coordinates is currently tightly coupled with classes that have lower levels of specialisation.
I don't want to alienate geographic-people from the user base, but I also don't want domain-specific code to get in the way of adding new features.

> If you really want something to show up without cartopy then I suggest that rather than implementing a projection to use, it would be better to plot in what you're calling native coordinates. i.e.., when plotting at fixed radius/depth/altitude without cartopy I'd prefer to have a plot of lat/lon on cartesian axes. I'm not sure how that fits in with what you're designing, though...

I can definitely do that in https://github.com/yt-project/yt/pull/4179
The follow up question would then be "how does one enable cartopy projections if they are off by default ?" and I need to clarify that I don't think just having cartopy installed should make a difference automatically, as this
- can create problems with reproducibility
- can be confusing for users as well as maintainers
- would not solve the code coupling issue I'm going at

If we can agree on this, there's an API choice to be made, and I don't have (other) strong opinions about it.

> And just a final note on the "transform" vs "projection" terminology -- I've thought before that it would be clearer to rename the data_transform attribute to be crs (coordinate reference system) in the GeopgraphicCoordinateHandler.

How about `coordinate_reference_system` so even newbies (myself included) know what it means ?
And for the record I'm all for it !

**Comment by chrishavlin** [comment]
> How about coordinate_reference_system
yes! makes sense to be more explicit.

> "how does one enable cartopy projections if they are off by default ?" 

that's a good question... how about a config option? `enable_cartopy` or something similar? If True, the behavior would be what yt now does (use use the transform (or `coordinate_reference_system`) and default projection to build a cartopy GeoAxes), if False plot in "native" coordinates. Would be easy to toggle on/off for testing and also nice from a user perspective to be able to turn it on and forget about it. But I could see it being kind of tricky to implement.

**Comment by neutrinoceros** [comment]
I'm slowly making my way back in this conversation, sorry for taking so long @chrishavlin !
For now I just want to triage this out of the 4.2 milestone as I'd rather not block the release over this.
